app/controllers/lti_launch_controller.rb

```diff
@@ -131,17 +132,64 @@ def validate_nrps_access
     # rubocop:enable Layout/LineLength
   end
 
+  def get_public_key(platform_public_key_pem, platform_public_jwks)
+    # import public key depending on whether we have a PEM format
+    # or list of JWKs
+    if !platform_public_key_pem.nil?
+      begin
+        rsa_public_key = OpenSSL::PKey::RSA.new(platform_public_key_pem)
+        return rsa_public_key
+      rescue StandardError
+        return nil
+      end
+    end
+    platform_public_jwks.each do |platform_public_key|
+      # JWT has a "key-id" header which specifies which public key to use
+      next unless platform_public_key["kid"] == @jwt[:header]["kid"]
+
+      begin
+        rsa_public_key = JWT::JWK::RSA.import(platform_public_key).public_key
+        return rsa_public_key
+      rescue StandardError
+        return nil
+      end
+    end
+    nil
+  end
+
   def validate_jwt_signature(id_token)
-    rsa_public = OpenSSL::PKey::RSA.new(Rails.configuration.lti_settings["platform_public_key"])
+    if !Rails.configuration.lti_settings["platform_public_key"].nil?
+      # static platform public key, so take key from yml
+      @platform_public_key_pem = Rails.configuration.lti_settings["platform_public_key"]
+    elsif !Rails.configuration.lti_settings["platform_public_jwks_url"].nil?
+      # fetch JWKS from provided keys URL
+      conn = Faraday.new(
+        url: Rails.configuration.lti_settings["platform_public_jwks_url"],
+        headers: { 'Content-Type' => 'application/json' }
+      )
+      # make a GET request to public JWK endpoint
+      response = conn.get("")
+      if response.body["keys"].nil?
+        LtiError.new("No keys were found from public JWK url", :internal_server_error)
+      end
+      @platform_public_jwks = JSON.parse(response.body)["keys"]
+    else
+      LtiError.new("No platform public key or public JWK url provided", :internal_server_error)
+    end
+    rsa_public_key = get_public_key(@platform_public_key_pem, @platform_public_jwks)
```

Comment:
Do we need instance variables here, given that only `get_public_key` uses them?

Reply:
Yeah it's unnecessary, made variables local